pipeline-scripts/buildlib.groovy

```diff
@@ -66,6 +66,12 @@ def registry_login() {
                       usernameVariable: 'USERNAME', passwordVariable: 'PASSWORD']]) {
         sh 'docker login -u $USERNAME -p $PASSWORD quay.io'
     }
+    // 2020-07-30 (https://issues.redhat.com/browse/ART-1193)
+    // Login to quay.io for push to quay.io/openshift-qe-optional-operator
```

Comment:
we can't login to quay.io twice with different credentials; you only get one at a time, and worse than that, by default all jobs share the same `~/.docker/config.json` where these are stored, so a job that logs in with one credential will break jobs that might be in the middle of using a different credential.

this isn't a good situation anyway, so i would like to fix it generically by isolating jobs to using their own docker config (we only use it for auth anyway). 

unfortunately i can't think of a clean way to accomplish this. we can configure docker by setting `env.DOCKER_CONFIG = env.PWD` for jobs that use docker config; but as far as I can see, `oc` doesn't heed a variable like this -- it requires a command-line arg, and that includes down in doozer which is what actually does this push.

so i'm afraid there is more difficulty here than i thought.

Reply:
Yeah, I make the doozer `--registry-config` flag to be exact same with `DOCKER_CONFIG` and it will automatically find the config files.